### ISSUE TITLE
aws-cdk-lib: Resolution error: Cannot find module '../../region-info/build-tools/fact-tables'

### ISSUE THREAD

**Issue by ajaylamba-provar** [aws/aws-cdk#31492]
Please add your +1 👍 to let us know you have encountered this
---

### Overview:

cdk build failure in stacks due to incorrect export of '../../region-info/build-tools/fact-tables'

#### Complete Error Message:

On running command `npx cdk synth`

Error: Resolution error: Resolution error: Resolution error: Cannot find module '../../region-info/build-tools/fact-tables'
Require stack:
`- /Documents/GitHub/test/node_modules/aws-cdk-lib/aws-ec2/lib/vpc-endpoint.js
- /Documents/GitHub/test/node_modules/aws-cdk-lib/aws-ec2/lib/index.js
- /Documents/GitHub/test/node_modules/aws-cdk-lib/aws-ec2/index.js
- /Documents/GitHub/test/lib/vpc-stack.ts`

Versions:
"aws-cdk-lib": "^2.159.0",

### Regression Issue

- [ ] Select this option if this issue appears to be a regression.

### Last Known Working CDK Version

2.158.0

### Expected Behavior

The synth operation should be successful without any error.

### Current Behavior

The synth command is getting failed with the error:
Error: Resolution error: Resolution error: Resolution error: Cannot find module '../../region-info/build-tools/fact-tables'
Require stack:

### Workaround:

Pin CDK version to `2.158.0`

### Solution:
Revert PR https://github.com/aws/aws-cdk/pull/31496

### Reproduction Steps

Create a VPC stack by using import statement as `import * as ec2 from 'aws-cdk-lib/aws-ec2';`

### CDK CLI Version

2.159.0

### Node.js Version

18.17.0

### OS

MacOS

### Language

TypeScript

**Comment by ajaylamba-provar** [comment]
After inspecting the node_modules folder of the project, I found that build-tools folder is not present inside `aws-cdk-lib/region-info`

**Comment by nelsonsilva-code** [comment]
I have also encountered this issue, and can confirm that this is only happening in 2.159.0, since my builds started breaking after the last package publishing.

**Comment by msciortino** [comment]
Same here

**Comment by shani-asad** [comment]
I'm also experiencing this issue

**Comment by ajaylamba-provar** [comment]
Pinning to version `2.158.0` works, thanks @nelsonsilva-code

**Comment by pahud** [comment]
Taking a look

**Comment by pahud** [comment]
Can't reproduce it.

`cdk init` a new empty project using 2.159.0 with

```ts
import * as cdk from 'aws-cdk-lib';
import * as ec2 from 'aws-cdk-lib/aws-ec2';
import { Construct } from 'constructs';
// import * as sqs from 'aws-cdk-lib/aws-sqs';

export class CdktestStack extends cdk.Stack {
  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    new ec2.Vpc(this, 'MyVpc');
  }
}
```

`npx cdk synth` works to me.

`region-info` exists as well

```
 % ls node_modules/aws-cdk-lib/region-info 
README.md       index.d.ts      index.js        lib
```

still investigating

**Comment by esteban-ricalde-mysa** [comment]
@pahud Yes, exactly the same error

**Comment by pahud** [comment]
OK reproducible.

```ts
export class Cdk2159Stack extends cdk.Stack {
  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    // create a random vpc endpoint for a new vpc
    const vpc = new ec2.Vpc(this, 'vpc', {
      maxAzs: 2,
      natGateways: 1,
    });
    const securityGroup = new ec2.SecurityGroup(this, 'sg', { vpc });
    securityGroup.addIngressRule(ec2.Peer.anyIpv4(), ec2.Port.tcp(443));
    const endpoint = vpc.addInterfaceEndpoint('endpoint', {
      service: ec2.InterfaceVpcEndpointAwsService.EC2,
      securityGroups: [securityGroup],
    });
  }
}
```

root cause could be - https://github.com/aws/aws-cdk/commit/95c49abdfa4ad77a0c0fcb82a230778dcc2ea59a

**Comment by shikha372** [comment]
Fix is now release as part of [v2.159.1](https://github.com/aws/aws-cdk/releases/tag/v2.159.1)
